addons/pos_event/static/src/app/components/popup/event_configurator_popup/event_configurator_popup.js

```diff
@@ -47,19 +47,40 @@ export class EventConfiguratorPopup extends Component {
             return this.props.availabilityPerTicket[ticket.id];
         }
         const ticketAvailability = this.props.availabilityPerTicket[ticket.id][this.slotId];
-        const existingUnsyncRegistration = this.pos.models["event.registration"].filter(
-            (r) => typeof r.id === "string" && r.event_slot_id.id === this.slotId
-        );
         if (ticketAvailability === "unlimited") {
             return ticket.event_id.seats_limited ? ticket.event_id.seats_available : "unlimited";
         }
-        return Math.max(ticketAvailability - existingUnsyncRegistration.length, 0);
+        return Math.max(ticketAvailability, 0);
+    }
+    getTicketRemainingQty(ticket) {
+        return Math.max(this.getTicketMaxQty(ticket) - this.state[ticket.id].qty, 0);
     }
     getProductProxy(productId) {
         return this.pos.models["product.product"].get(productId);
     }
     confirm() {
         const data = [];
+        // Check ticket quantity based on event or slot limitations
+        const event = this.props.tickets[0].event_id;
+        const slotOrEventAvailableSeats = this.slotId
+            ? this.slotAvailability
+            : event.seats_available;
+        const totalTicketQuantity = (Object.entries(this.state) ?? []).reduce(
+            (sum, [_, { qty }]) => sum + qty,
+            0
+        );
```

Comment:
In theory, still works if we write something else in the state? (for instance if no `qty` is found in the entry?)